tenacity/_asyncio.py

```diff
@@ -17,35 +17,31 @@
 
 import functools
 import sys
-import typing
-from asyncio import sleep
+from asyncio import sleep as aio_sleep
+from inspect import iscoroutinefunction
+from typing import Any, Awaitable, Callable, TypeVar, Union
 
-from tenacity import AttemptManager
-from tenacity import BaseRetrying
-from tenacity import DoAttempt
-from tenacity import DoSleep
-from tenacity import RetryCallState
+from tenacity import AttemptManager, BaseRetrying, DoAttempt, DoSleep, RetryAction, RetryCallState, TryAgain
 
-WrappedFn = typing.TypeVar("WrappedFn", bound=typing.Callable)
-_RetValT = typing.TypeVar("_RetValT")
+WrappedFn = TypeVar("WrappedFn", bound=Callable)
+_RetValT = TypeVar("_RetValT")
 
 
 class AsyncRetrying(BaseRetrying):
-    def __init__(self, sleep: typing.Callable[[float], typing.Awaitable] = sleep, **kwargs: typing.Any) -> None:
+    def __init__(self, sleep: Callable[[float], Awaitable] = aio_sleep, **kwargs: Any) -> None:
```

Comment:
There are too many changes that are unrelated to this PR. 
Could you avoid changing code that does not need to be changed?

Reply:
It was giving warnings because variable and function had same names (`sleep`), should I make another PR for this?

Reply:
Also, writing `typing.` before types every time made some lines too long, and it was looking unnecessary, should I revert that one too?

Reply:
sorry for late response, I will try to find time in upcoming days and split this PR into 2 different PRs.

Reply:
I created a new PR with AsyncRetries related changes #363 
Will make another PR for coding style improvements after that.

Closing this PR (will use this branch as template)